core/modules.rs

```diff
@@ -1760,12 +1769,16 @@ import "/a.js";
       (mod_a, mod_b)
     };
 
-    runtime.instantiate_module(mod_b).unwrap();
+    global_realm
+      .instantiate_module(runtime.v8_isolate(), mod_b)
+      .unwrap();
```

Comment:
A lot of these changes seem superficial - how about we still keep all of these methods on the `JsRuntime` and they still internally call into the "global realm"? I believe in 99% cases we won't be handling multiple realms and it just makes the common path more complicated.

Reply:
Using the global realm when a different realm should be used might lead to crossing objects from the global realm into the `ShadowRealm`, or vice versa, allowing prototype pollution across the `ShadowRealm` boundary. Requiring these methods (as well as `handle_scope`, `execute_script`, etc.) to be accessed through a `JsRealm` forces you to think about which realm you need to work with, and also draws reviewers' attention to it.

Reply:
> Using the global realm when a different realm should be used might lead to crossing objects from the global realm into the ShadowRealm, or vice versa

How is that possible? I thought that if you try to act on an object from a different context you'd get a V8 panic

Reply:
You get a panic for different isolates, not contexts. Although V8 keeps track of [in which context an object was created](https://docs.rs/v8/latest/v8/struct.Object.html#method.get_creation_context), this isn't really a thing in the spec, and in browsers you can use an object from another context just fine with iframes (`iframe.contentWindow.document`, for example).

Reply:
Okay, seems like a valid reason to have separate API then.